web/components/MiniRoadmap.test.tsx

```diff
@@ -1,22 +1,36 @@
-import { fireEvent, render, RenderResult } from "@testing-library/react";
+import { fireEvent, render, RenderResult, within } from "@testing-library/react";
 import { MiniRoadmap } from "./MiniRoadmap";
 
-import { generateRoadmap, generateStep, generateTask } from "@/test/factories";
+import { generateStep, generateTask } from "@/test/factories";
 import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
 
 jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
 
 describe("<MiniRoadmap />", () => {
   let subject: RenderResult;
   beforeEach(() => {
-    useMockRoadmap(
-      generateRoadmap({
-        steps: [
-          generateStep({ name: "step1", tasks: [generateTask({ name: "task1", id: "task1" })] }),
-          generateStep({ name: "step2", tasks: [generateTask({ name: "task2", id: "task2" })] }),
-        ],
-      })
-    );
+    useMockRoadmap({
```

Comment:
It's an opinion, but I have a preference as tests grow bigger to keep specific setup cases within the test block itself instead of the beforeEach. Like, so that each test contains a given-when-then when you read it, and you don't have to scroll all the way up to the beforeEach to understand the "given" setup

Used to get in heated debates about this opinion with a friend on a previous team though :) so I understand it's not shared by all. what do y'all think of having the useMockRoadmap call in your specific test? Since none of the rest of the tests care about sections, having them listed out overall at the top is clutter for everyone except our specific test. And then you have to _know_ which pieces are for which, ie that the `tasks` properties are meaningful to the first 3 tests but not the last one, and the `section` properties are meaningful only to the last test. So it might be more clear to just define what we need for each test/set